range-diff.c

```diff
@@ -583,16 +625,34 @@ int show_range_diff(const char *range1, const char *range2,
 	if (range_diff_opts->left_only && range_diff_opts->right_only)
 		res = error(_("options '%s' and '%s' cannot be used together"), "--left-only", "--right-only");
 
+	/* Check range sizes early before reading patches */
+	if (!res) {
+		count1 = count_range_commits(range1);
+		count2 = count_range_commits(range2);
```

Comment:
Would probably be more elegant to integrate that counting into `read_patches()` rather than forcing every revision walk to happen twice, in separate processes.